lib/index.js

```diff
@@ -208,14 +208,25 @@ function validateDate(date, key) {
     return date;
   }
   if(date instanceof Date) {
-    return (Math.round(date.getTime() / 1000));
-  } else if(Number.isInteger(date)) {
+    const timestamp = date.getTime() / 1000;
+    if(key === 'created') {
+      // using Math.ceil here can result
+      // in timestamps that are 1 second in the future
+      return Math.floor(timestamp);
```

Comment:
For created this does need to be `Math.ceil` so we don't create signatures that fall outside of the requested range. For default `created` behavior (where the user isn't passing in the date, we'll want to use Math.floor).

Reply:
this results in an error where the timestamp is 1 second before now.

Reply:
Addressed here: https://github.com/digitalbazaar/http-signature-header/pull/17/commits/470a57786c0b05a4a310574ef484c1e616381e05

Reply:
I think we're miscommunicating. If `created` is passed into the main API, it should be rounded down by this helper. If `created` is auto-generated (because it wasn't passed into the API), it should use `Math.floor(Date.now() / 1000)`.